IUBLibTech/ngao
# Use integers instead of strings for years

Refs [AR-91](https://bugs.dlib.indiana.edu/browse/AR-91)

# Summary
The blacklight range-limit gem expects integers instead of strings,
however, NGAO was indexing it's years as strings.  This resulted in
lexical rather than numerical sorting of years.

# References
See https://github.com/projectblacklight/blacklight_range_limit#requirements

# Fix
This change modifies the Year value from being stored as a string
to storing it as an integer  by changing the date_range field
from _sim (string,  indexed, multivalued)
 to  _iim (integer, indexed, multivalued)

# Date sort before
* 1
* 115
* 1012
* 1995
* 2022
* 214
* 895

# Date sort after
* 1
* 115
* 214
* 896
* 1012
* 1995
* 2022

# Side Effects
Because this change impacts the Solr schema, it requires a patch to the Solr docker container.  This PR implements that via a modification to the CircleCI config file.  A more permanent solution would be to mint a new Solr image and remove the changes to `./.circleci/config.yml` 

# Testing / Reproduction instructions
1. Do a search for all records in NGAO
2. Open the date range (year) facet
3. Type the dates "200" and "300" in the beginning and ending date box
4. You should see less that 20 matching collections (prior to the change, this range would lexically include 20xx years and returns 80,000+ results)

# Deployment
Once this change is deployed, the entire data set will need to be re-indexed to ensure date range data is indexed numerically rather than lexically.
